### PR TITLE
document Vertex macro loading and modifying

### PR DESCRIPTION
**Pull request by coco875**

### CODE REVIEW

**Comment by ariahiro64** [comment]
vertex value flag tells the microcode to use packed normals in f3dex3 so it does have a big meaning

**Comment by coco875** [comment]
I just copy from n64 official doc can you give a phrase to replace with I don't know so much this feature

**Comment by coco875** [comment]
@ariahiro64 I change it with what somone suggest in the discord it's correct now ?

File: gbi.h [hunk 1]
```diff
@@ -252,10 +252,28 @@ longer a multiple of 8 (DMA word). This was not used in any command anyway. */
...
+ * @brief changes the S and T values (texture coordinates of the vertex). The high 16 bits of val specify the S coordinate, and the low 16 bits specify the T coordinate. Each coordinate is an S10.5 number.
+ * 
+ */
 #define G_MWO_POINT_ST           0x14
-#define G_MWO_POINT_XYSCREEN     0x18 /* not recommended to use, won't work if */
-#define G_MWO_POINT_ZSCREEN      0x1C /* the tri gets clipped */
+/**
+ * @brief change the screen coordinates of the vertex. The high 16 bits of val specify the X coordinate and the low 16 bits specify the Y coordinate. Both coordinates are S13.2 numbers with 0,0 being the upper-left of the screen, positive X going right, and positive Y going down.
+ * 
+ * not recommended to use, won't work if
```
Comment (sauraen) [inline gbi.h]: The end of this comment got moved to the next macro. It was originally a multiline comment across two definitions.

File: gbi.h [hunk 2]
```diff
@@ -2248,11 +2266,31 @@ _DW({                                                   \
...
 /**
- *        +--------+----+---+---+----+------+-+
- *  G_VTX |  cmd:8 |0000|  n:8  |0000|v0+n:7|0|
- *        +-+---+--+----+---+---+----+------+-+
- *        | |seg|         address             |
- *        +-+---+-----------------------------+
+ * @brief macro which loads an internal vertex buffer in the RSP with points that are used by @ref gSP1Triangle macros to generate polygons at the end display list.
+ * 
+ * 
+ * It loads an internal vertex buffer in the RSP with points that are used by @ref gSP1Triangle macros to generate polygons. This vertex cache can hold up to 16 vertices, and the vertex loading can begin at any entry (index) within the cache. The vertex coordinates (x,y,z) are encoded in signed 2's complement, 16-bit integers. The texture coordinates (s,t) are encoded in S10.5 format. A vertex either has a color or a normal (for shading). These values are 8-bit values. The colors and alphas are treated as 8-bit unsigned values (0-255), but the normals are treated as 8-bit signed values (-128 to 127). Therefore, the appropriate member of the union to use (.v. or .n.) depends on whether you are using colors or normals.
```
Comment (sauraen) [inline gbi.h]: Change 16 to `56 (F3DEX3)`

File: gbi.h [hunk 3]
```diff
@@ -2248,11 +2266,31 @@ _DW({                                                   \
...
+ * 
+ * The coordinates (x,y,z) are transformed using the current 4x4 projection and model view matrices, and (s,t) are transformed using the scale defined by gSPTexture.
+ * 
+ * # Example
+ * To load vertex cache entry 2,3,4, use this code:
+ * ```c
+ * gSPVertex(glistp++, v, 3, 2);
+ * ```
+ * 
+ * The range of n or v0 has been changed along with the change in vertex cache size. The number of vertices which can be loaded at once is less than 56, so it is necessary to split the load when loading more than 56 vertices. Please use gSPVertex multiple times when loading more than 56 vertexes.
```
Comment (sauraen) [inline gbi.h]: Please remove this line, this was SGI documentation for a limitation which was removed long before F3DEX3 (where you could load fewer vertices at a time than you could store in total). EX3 has 56 verts and you can load any set of them at once up to all 56.

File: gbi.h [hunk 4]
```diff
@@ -2248,11 +2266,31 @@ _DW({                                                   \
...
+ * 
+ * @note
+ * Because the RSP geometry transformation engine uses a vertex list with triangle list architecture, it is quite powerful. A simple one-triangle macro retains maximum performance.
```
Comment (sauraen) [inline gbi.h]: I don't know what SGI was exactly comparing this to, but the statement `A simple one-triangle macro retains maximum performance.` is not true today, that is the least performant compared to SP2Triangles or the new 5 tris commands in EX3.

File: gbi.h [hunk 5]
```diff
@@ -2248,11 +2266,31 @@ _DW({                                                   \
...
+ * 
+ * @param v is the pointer to the vertex list (segment address)
+ * @param n is the number of vertices
+ * @param v0 is the load vertex by index vo(0~15) in vertex buffer
```
Comment (sauraen) [inline gbi.h]: Please update this to 0-55

File: gbi.h [hunk 6]
```diff
@@ -3073,11 +3113,53 @@ typedef union {
...
+ * gSPModifyVertex(glistp++, 3, G_MWO_POINT_ST, 0x00600050);
+ * 
+ * // load texture of triangle 2
+ * gDPLoadTextureBlock(...);
+ * 
+ * // draw triangle 2 using vertex #3
+ * gSP1Triangle(glistp++, 1,2,3,0);
+ * ```
+ * 
+ * @param vtx specifies which of the RSP's vertices (0-15) to modify
```
Comment (sauraen) [inline gbi.h]: Please update this to 0-55

File: gbi.h [hunk 7]
```diff
@@ -3073,11 +3113,53 @@ typedef union {
...
 
 /**
- * Insert values into Vertices
- *
- * vtx = vertex number 0-55
- * where = which element of point to modify (byte offset into vertex)
- * num   = new value (32 bit)
+ * @brief You can use this macro to modify certain sections of a vertex after it has been sent to the RSP (by the gSPVertex macro).
+ * 
+ * This is an advanced macro. You need a good understanding of how vertices work in the RSP microcode before you use this macro (refer to gSPVertex). The range of vtx has been changed with as the vertex cache increased. Its range is now the same as that for gSPVertex.
```
Comment (sauraen) [inline gbi.h]: Please remove `The range of vtx has been changed with as the vertex cache increased. Its range is now the same as that for gSPVertex.`, this is again referring to old limitations pre-F3DEX.

**Review by sauraen — CHANGES_REQUESTED** [review]

**Comment by coco875** [comment]
I fix all comment you made I don't know so much about F3DEX this include modification you made for F3DEX3 (for 56 vertices I just forget x))

**Review by sauraen — APPROVED** [review]

**Pull request merged**